source/rules.txt

```diff
@@ -11,7 +11,7 @@ Define Data Access Permissions
    
    Role-based Permissions </rules/roles>
    Rule Expressions </rules/expressions>
-   Filter Incoming Queries </rules/filters>
+   Filter Query Results </rules/filters>
```

Comment:
Why the name change? These filters are the first step in the rules process but IMO the new name implies they're the final step.

Reply:
You're right that the end user will see the results of a Filter in the query result set that's returned to them.

In practice though the purpose of these is more to optimize document-level rule evaluation because it's a fairly expensive operation (i.e. it slows down your query) and it scales with the number of documents it needs to evaluate. So rather than evaluating rules for everything that matches an arbitrary user query, you can say "I _know_ that I can always filter out these additional documents under these conditions because they won't be returned to the user anyway."

That said, there may be a use case that's closer to "the end user will see the results of a Filter in the query result" for Device Sync rules that I haven't really thought about.

Either way, we may want to eventually reframe this around optimizing rule execution (or make a new section/page about it).

Reply:
Thanks! I see that the focus is more at the beginning of the process, on optimizing the query. I'll change the name back @nlarew